#3457 increase max length of VAR* types

grandinj opened the pull request.

grandinj committed:
#3457 increase max length of VAR* types

katzyn commented on h2/src/main/org/h2/engine/Constants.java:
```diff
@@ -267,8 +267,11 @@
+     * encoding data fields when storing rows.
      */
-    public static final int MAX_STRING_LENGTH = 1024 * 1024;
+    public static final int MAX_STRING_LENGTH = 1024 * 1024 * 1024;
```
It should be `1_000_000_000` as specified in documentation.

1024^3 is actually larger than actual limit of `java.lang.String` since Java 9 (for strings with at least one character with code larger than 255).

grandinj committed:
fix constant

to match the new documented value

grandinj committed:
reduce re-allocation in getListaggTruncate

which should speed it up

grandinj committed:
remove LISTAGG oiverflow tests

there is no way to test this without using up so much memory we'd probably get flagged for abusing CI

grandinj committed:
fix overflow

grandinj committed:
fix tests

grandinj committed:
disable LOB testLimits test

causes OOM on CI
